### PR TITLE
[Core] Add ray check-open-ports cli

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Follow-up of https://www.anyscale.com/blog/update-on-ray-cve-2023-48022-new-verification-tooling-available

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_cli.py [hunk 1]
```diff
@@ -979,6 +983,79 @@ def output_ready():
...
+                        "checked_ports": payload["ports"],
+                    }
+                ).encode("utf-8")
+            )
+
+    server = HTTPServer(("127.0.0.1", 9000), OpenPortCheckServer)
+    server_thread = threading.Thread(target=server.serve_forever)
+    server_thread.start()
+
+    yield OpenPortCheckServer
```
Comment (thomasdesr) [inline python/ray/tests/test_cli.py:1013]:
```suggestion
    server = HTTPServer(("127.0.0.1", 0), OpenPortCheckServer)
    server_thread = threading.Thread(target=server.serve_forever)
    server_thread.start()

    yield OpenPortCheckServer, f"http://{server.server_address[0]}:{server.server_address[1]}"
```

I'd probably do something like this that way we don't have port conflicts if there's two of these running (e.g. if we parallelize this test)

File: python/ray/util/check_open_ports.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        for node_id in ray_node_ids
+    }
+
+    results = []
+    for node_id, per_node_task in per_node_tasks.items():
+        try:
+            results.append(ray.get(per_node_task))
+        except Exception as e:
+            cli_logger.info(f"Failed to check on node {node_id}: {e}")
+    return results
```
Comment (thomasdesr) [inline python/ray/util/check_open_ports.py:113]: nit:

```suggestion
            cli_logger.info(f"Failed to check on node {node_id}: {e}")

    return results
```

**Review by thomasdesr — APPROVED** [review]
Suggestion to improve the tests, but otherwise lgtm!

**Comment by can-anyscale** [comment]
likely broke https://github.com/orgs/anyscale/projects/76/views/1?pane=issue&itemId=58642206, i'm reverting to unblock

**Pull request merged**